pkg/api/registered/registered.go

```diff
@@ -25,71 +26,97 @@ import (
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
-// List of registered API versions.
-// The list is in the order of most preferred to the least.
-var RegisteredGroupVersions []unversioned.GroupVersion
+var (
+	// registeredVersions represents all enabled API versions. Please call
+	// RegisterVersions() to register the versions.
+	registeredVersions = map[unversioned.GroupVersion]struct{}{}
```

Comment:
> What do you think? Why do we need to keep record of the versions that are compiled but not enabled?

I think we might want to know all the available versions even when a subset is chosen.  It would allow us to do things like look at a whitelist (env var), compare against the available, and tell them which versions are disabled.  There might be other ways to do it, but I think the approach you've laid out in this pull of unconditional registration followed by inspection of enablement will make it easier to reason about the code and debug it after the fact.